### PR TITLE
Implement `LinkedEventIntake`

### LINKED ISSUE #9525: Create a version of `EventIntake` that is compatible with the new intake pipeline
The current `EventIntake` class includes elements that aren't relevant to the new intake pipeline, which is being implemented under the new platform framework.

A new version of `EventIntake` should be created, with the irrelevant aspects removed. `EventIntake` should remain unchanged, to allow the new intake pipeline to be fully hidden behind a feature flag

### PR DESCRIPTION
**Pull request by alittley**
- closes #9525

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/LinkedEventIntake.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        // find all events that just became ancient and did not reach consensus, these events will be considered stale
+        final Collection<EventImpl> staleEvents = shadowGraph.findByGeneration(
+                previousGenerationNonAncient,
+                consensusSupplier.get().getMinGenerationNonAncient(),
+                LinkedEventIntake::isNotConsensus);
+
+        for (final EventImpl staleEvent : staleEvents) {
+            staleEvent.setStale(true);
+            dispatcher.staleEvent(staleEvent);
+            logger.warn(STALE_EVENTS.getMarker(), "Stale event {}", staleEvent::toShortString);
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/LinkedEventIntake.java:195]: This is either test only logging (most likely the case) or a spam risk.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/LinkedEventIntake.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        }
+    }
+
+    /**
+     * Returns true if the event has not reached consensus
+     *
+     * @param event the event to check
+     * @return true if the event has not reached consensus
+     */
+    private static boolean isNotConsensus(final EventImpl event) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/LinkedEventIntake.java:205]:
```suggestion
    private static boolean isNotConsensus(@NonNull final EventImpl event) {
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/LinkedEventIntake.java (hunk 1 above)
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/LinkedEventIntake.java:195]: Would you prefer removal, or rate limited logging?

Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/LinkedEventIntake.java:195]: My preference is removal.

**Review by cody-littley — APPROVED** [review]

**Pull request merged**